src/main/java/kostka/moviecatalog/service/RedisService.java

```diff
@@ -0,0 +1,41 @@
+package kostka.moviecatalog.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class RedisService {
+    static final String LATEST_MOVIES_KEY = "latestMovies";
+    private static final Long N_LATEST_MOVIES = 5L;
+
+    private static final Logger LOGGER = LogManager.getLogger("CONSOLE_JSON_APPENDER");
+
+    private RedisTemplate<String, String> redisTemplate;
+
+    @Autowired
+    public RedisService(final RedisTemplate<String, String> redisTemplate) {
+        this.redisTemplate = redisTemplate;
+    }
+
+    public void updateLatestMovie(final String movieId) {
+        LOGGER.info("Adding movieId '{}' to the redis cache", movieId);
+        while (redisTemplate.opsForList().size(LATEST_MOVIES_KEY) >= N_LATEST_MOVIES) {
```

Comment:
Tohle by se možná celé dalo nahradit tím, že vložíš prostě ID do seznamu a pak zavoláš LTRIM nebo jak se to jmenuje u RedisTemplate. To ti nechá v seznamu jen určitý počet položek.

Příklad:
[list = 3 2 1]
lpush 4
[list = 4 3 2 1]
ltrim 3
[list = 4 3 2]

Reply:
skvely tip, díky!